### DISCUSSION TITLE
[BUG] JSQLParser 4.5/4.6 : RDBMS : extension field exception

### DISCUSSION THREAD

**Discussion by licai1210** [JSQLParser/JSqlParser#1766] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/12244247/232457802-0f38726c-cc1f-4b37-a2b9-e3b0718125a8.png)

Hello, this issue is normal in 4.4, and there is an inaccurate issue with dynamically adding fields in the insert statement in 4.5/4.6. May I know how to solve it? Thank you

**Comment by manticore-projects** [comment]
Greetings.

A lots of API has changed between 4.4 and 4.6 in order to streamline `VALUES (..)` statements. 
The easiest way to figure out the AST is using [JSQLFormatter](http://217.160.215.75:8080/jsqlformatter/JSQLFormatter/demo.html), which can visualize the Tree of Java Objects:

![image](https://user-images.githubusercontent.com/18080123/232459255-09f9f051-49c3-4769-87f2-a8aaa6040443.png)

[You can try it online here](http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20JIOQygogSgKgBKGB5OBbAngFwIYCMA2ApnABQBQclcAxgPb4CMcAlGQGoCCAMgKoRik4TVgG4gA)

**Comment by licai1210** [comment]
Hello, I need to add field values to existing SQL statements. May I know how to proceed? Thank you

**Comment by manticore-projects** [comment]
Recommendation: Checkout PR #1754 which will allow for
```java
    @Test
    void testIssue1765() {
        String sqlStr = "INSERT INTO mytable (col1)\n"
                        + "VALUES ( 1 )";

        Insert insert = new Insert()
                            .withTable( new Table("mytable") )
                            .withColumns( Arrays.asList(new Column("col1")) )
                            .withSelect( new Values().addExpressions(new LongValue(1)) );

        assertStatementCanBeDeparsedAs(insert, sqlStr,true);
    }
```

**Comment by manticore-projects** [comment, marked as answer]
If you use Release 4.6 then your code will look like this:
```java
    @Test
    void testIssue1765() {
        String sqlStr = "INSERT INTO mytable (col1)\n"
                        + "VALUES ( 1 )";

        Select select = new Select()
                                .withSelectBody( new ValuesStatement().addExpressions( new LongValue(1)) );

        Insert insert = new Insert()
                                .withTable( new Table("mytable") )
                                .withColumns( Arrays.asList(new Column("col1")) )
                                .withSelect(  select );

        assertStatementCanBeDeparsedAs(insert, sqlStr,true);
    }
```
